Add counts_hist function

rameshvs opened the pull request:
[x] Wrote test for feature
[x] Added changes in the Changelog section in README.md
[x] Bumped version number (delete if unneeded)

**Changes proposed:**
Add `hist_of_counts` function, as in https://github.com/data-8/datascience/issues/318.

rameshvs commented:
@papajohn @davidwagner I can't seem to request a review or assign anyone, but here's the PR for #318 .

papajohn commented on datascience/tables.py:
```diff
@@ -2569,6 +2569,102 @@ def draw_hist(values_dict):
+        3     | 2
+        3     | 2
+        1     | 10
+        >>> t.hist() # doctest: +SKIP
```
Is this meant to be `t.hist_of_counts()`, or is it just stale and should be deleted?

papajohn commented on datascience/tables.py:
```diff
@@ -2569,6 +2569,102 @@ def draw_hist(values_dict):
+
+        # TODO test case bins = np.arange(3, 40, 5/3)
+        # TODO test case bins = np.arange(1e-14, 6e-14, 1e-13)
+        # TODO test case bins = np.arange(1e-20, 2e-20, 1e-21)
```
Looks like some of these are done already.

rameshvs commented:
Thanks for the catch! Comments addressed. I also changed one of the docstring examples to use counts, and changed it so that when using `bin_column`, it checks whether the counts are integers (or close to integers). 

The check only works if columns are passed in: fixing it would probably require a bit of refactoring to avoid duplication between `hist` and `hist_of_counts`.
